File: public/examples/integrations/mcp-servers/posthog_api/retry_batch_export_run_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "PosthogApi.RetryBatchExportRun"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'mode': 'execute',
    'batch_export_identifier': '123e4567-e89b-12d3-a456-426614174000',
    'batch_export_run_id': '456e7890-e12b-34d5-b678-426614174001',
    'project_id': '789e0123-e45f-67g8-h901-426614174002',
    'request_body': '{"status":"retry"}'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/posthog_api/retry_batch_export_run_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "PosthogApi.RetryBatchExportRun";

// Start the authorization process
const authResponse = await client.tools.authorize({
  tool_name: TOOL_NAME,
  user_id: USER_ID,
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "mode": "execute",
  "batch_export_identifier": "123e4567-e89b-12d3-a456-426614174000",
  "batch_export_run_id": "456e7890-e12b-34d5-b678-426614174001",
  "project_id": "789e0123-e45f-67g8-h901-426614174002",
  "request_body": "{\"status\":\"retry\"}"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
